Packs/AzureLogAnalytics/Integrations/AzureLogAnalytics/AzureLogAnalytics.py

```diff
@@ -94,32 +90,34 @@ def http_request(self, method, url_suffix=None, full_url=None, params=None,
 ''' INTEGRATION HELPER METHODS '''
 
 
-def format_query_table(table):
+def validate_params(refresh_token, managed_identities_client_id, client_credentials, enc_key, self_deployed,
+                    certificate_thumbprint, private_key):
+    if not refresh_token:
+        raise DemistoException('Token / Tenant ID must be provided.')
+    if not managed_identities_client_id:
+        if client_credentials and not enc_key:
+            raise DemistoException("Client Secret must be provided for client credentials flow.")
+        elif not self_deployed and not enc_key:
+            raise DemistoException('Key must be provided. For further information see '
+                                    'https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication')  # noqa: E501
+        elif not enc_key and not (certificate_thumbprint and private_key):
+            raise DemistoException('Key or Certificate Thumbprint and Private Key must be provided.')
```

Comment:
```suggestion
            raise DemistoException('Please provide either Key, Certificate Thumbprint or Private Key.')
```

Just making sure- we ask for one of the 3? ( not (x and y) == not x or not y )

Reply:
`x or (y and z)`
https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication#client-credentials